### ISSUE TITLE
new digital input driver: software busy read

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#2081]
ICU can only generate events on 8 pins, PAL can only generate events on 16 pins - that's just too little options

A 100% software input capture driver could be a thread reading from all specified pins in a constant loop, with lowest possible thread priority

**Comment by shadowm60** [comment]
i think here we would need to define the frequency range we should be able to read, since usually frequencies are not that high, we can read pins in a while look (cyclic value would be predefined).

**Comment by mck1117** [comment]
Yes, not an unreasonable idea.  Running on a super low priority thread will have (in theory) zero impact on anything else running already on the ECU.
